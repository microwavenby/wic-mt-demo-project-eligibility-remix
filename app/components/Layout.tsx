import { Trans } from "react-i18next";
import { Image } from "remix-image";
import { ReactElement } from "react";
import { Alert } from "@trussworks/react-uswds";
import TransLinks from "./TransLinks";

type Props = {
  children: ReactElement;
  demoMode?: string;
  missingData?: string;
};

const Layout = ({ children, demoMode, missingData }: Props): ReactElement => {
  return (
    <div className="container">
      {demoMode === "true" ? (
        <Alert type="warning" headingLevel="h6" slim={true} role="status">
          <TransLinks
            i18nTextKey={"demoAlertBanner.text"}
            i18nLinkKey={"demoAlertBanner.links"}
          />
        </Alert>
      ) : (
        ""
      )}
      <header className="header usa-header usa-header--basic" role="banner">
        <div className="usa-navbar">
          <div className="grid-row">
            <div className="desktop:grid-col-8">
              <div className="usa-logo margin-left-2">
                <em className="usa-logo__text">
                  <Trans i18nKey="Layout.header" />
                </em>
              </div>
            </div>
          </div>
        </div>
      </header>
      <main className="main">
        <div className="grid-row">
          <div className="desktop:grid-col-8 padding-2 padding-bottom-8">
            {missingData === "true" ? (
              <Alert
                type="error"
                headingLevel="h4"
                className="margin-bottom-3"
                role="alert"
              >
                <Trans i18nKey={"routingError"} />
              </Alert>
            ) : (
              ""
            )}
            {children}
          </div>
        </div>
      </main>
      <footer className="footer usa-footer usa-footer--slim">
        <div className="usa-footer__primary-section">
          <div className="grid-row">
            <div className="desktop:grid-col-8 padding-2">
              <div className="logos">
                <Image
                  src="/img/wic-logo.svg"
                  alt="WIC logo"
                  width={64.52}
                  height={32}
                />
                <Image
                  src="/img/montana-logo.svg"
                  alt="Montana DPHHS logo"
                  width={46.22}
                  height={32}
                />
              </div>
              <div className="font-body-3xs">
                <p>
                  <TransLinks
                    i18nTextKey="Layout.footer1.text"
                    i18nLinkKey="Layout.footer1.links"
                  />
                </p>
                <p>
                  <TransLinks
                    i18nTextKey="Layout.footer2.text"
                    i18nLinkKey="Layout.footer2.links"
                  />
                </p>
              </div>
            </div>
          </div>
        </div>
      </footer>
    </div>
  );
};

export default Layout;
