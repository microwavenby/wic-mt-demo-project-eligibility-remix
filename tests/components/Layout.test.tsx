import Layout from "app/components/Layout";
import { renderWithRouter } from "tests/helpers/setup";

import { testAccessibility } from "../helpers/sharedTests";

it("should match snapshot", () => {
  const { container } = renderWithRouter(
    <Layout children={<h1>'child'</h1>} />
  );
  expect(container).toMatchSnapshot();
});

it("should pass accessibility scan", async () => {
  await testAccessibility(<Layout children={<h1>'child'</h1>} />);
});
