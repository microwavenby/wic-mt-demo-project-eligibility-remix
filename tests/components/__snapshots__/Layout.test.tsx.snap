// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <header
      class="header usa-header usa-header--basic"
      role="banner"
    >
      <div
        class="usa-navbar"
      >
        <div
          class="grid-row"
        >
          <div
            class="desktop:grid-col-8"
          >
            <div
              class="usa-logo margin-left-2"
            >
              <em
                class="usa-logo__text"
              >
                Apply for WIC in Montana
              </em>
            </div>
          </div>
        </div>
      </div>
    </header>
    <main
      class="main"
    >
      <div
        class="grid-row"
      >
        <div
          class="desktop:grid-col-8 padding-2 padding-bottom-8"
        >
          <h1>
            'child'
          </h1>
        </div>
      </div>
    </main>
    <footer
      class="footer usa-footer usa-footer--slim"
    >
      <div
        class="usa-footer__primary-section"
      >
        <div
          class="grid-row"
        >
          <div
            class="desktop:grid-col-8 padding-2"
          >
            <div
              class="logos"
            >
              <img
                alt="WIC logo"
                class="remix-image"
                decoding="async"
                height="32"
                loading="lazy"
                src="/img/wic-logo.svg"
                style="min-width: 0; min-height: 0; background-size: cover; background-position: center;"
                width="64.52"
              />
              <img
                alt="Montana DPHHS logo"
                class="remix-image"
                decoding="async"
                height="32"
                loading="lazy"
                src="/img/montana-logo.svg"
                style="min-width: 0; min-height: 0; background-size: cover; background-position: center;"
                width="46.22"
              />
            </div>
            <div
              class="font-body-3xs"
            >
              <p>
                This site is a demonstration project built with 
                <a
                  class="usa-link usa-link--external"
                  href="https://dphhs.mt.gov/ecfsd/wic/index"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Montana WIC
                </a>
                 and is for example purposes only. If you need to submit an application for WIC, please use 
                <a
                  class="usa-link usa-link--external"
                  href="https://www.signupwic.com/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  SignUpWIC.com
                </a>
                 to find a location near you.
              </p>
              <p>
                This institution is an equal opportunity provider. 
                <a
                  class="usa-link usa-link--external"
                  href="https://www.fns.usda.gov/civil-rights/usda-nondiscrimination-statement-other-fns-programs"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  USDA Non-Discrimination Statement
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
